### PR TITLE
refactor labels to focus for 508 focusing bug

### PR DESCRIPTION
**Pull request by hughes-dustin**
## Description
https://github.com/department-of-veterans-affairs/va.gov-team/issues/1251

## Acceptance criteria
- [ ] As a screen reader (NVDA) user, I want to be able to press ENTER on the label and have focus set on the related input.
- [ ] As this user, I want to be able to focus the Learn more button as a separate element, and press ENTER or SPACE to open the modal window.

## Definition of done
- [ ] Events are logged appropriately
- [ ] Documentation has been updated, if applicable
- [ ] A link has been provided to the originating GitHub issue (or connected to it via ZenHub)
- [ ] No sensitive information (i.e. PII/credentials/internal URLs/etc.) is captured in logging, hardcoded, or specs

### CODE REVIEW

**Review by zurbergram — APPROVED** [review]
LGTM

**Review by delli-gatti-michael — APPROVED** [review]
Approved by QA

File: src/applications/gi/utils/render.jsx [hunk 1]
```diff
@@ -33,18 +33,26 @@ export const renderPreferredProviderFlag = result => {
...
+          type="button"
+          className="va-button-link learn-more-button"
+          onClick={showModal.bind(component, modal)}
+        >
+          (Learn more)
+        </button>
+      </span>
+    );
+  }
+  return <label htmlFor={name}> {text} </label>;
```
Comment (cvalarida) [inline src/applications/gi/utils/render.jsx]: Looks like this function solves two different problems:
1) Render the `(Learn more)` modal toggle
2) Render a simple `<label>`

Does this need to be in here? If it makes sense to have this wrapped in a function, I recommend putting it in its own function, separate from the `(Learn more)` modal.

File: src/applications/gi/utils/render.jsx [hunk 2]
```diff
@@ -33,18 +33,26 @@ export const renderPreferredProviderFlag = result => {
...
-    <button
-      type="button"
-      className="va-button-link learn-more-button"
-      onClick={showModal.bind(component, modal)}
-    >
-      (Learn more)
-    </button>
-  </span>
-);
+export const renderLabel = ({ name, text, modal, showModal, component }) => {
```
Comment (cvalarida) [inline src/applications/gi/utils/render.jsx]: Can you make sure this is unit tested?

File: src/applications/gi/utils/render.jsx (hunk 1 above)
Comment (hughes-dustin) [reply src/applications/gi/utils/render.jsx]: I was trying to write in a way that allows us to pass in our label properties for either a `learn more` or regular label  in `gi/components/Dropdown.jsx` component without having to create a flag at the component level. If it makes sense to move the logic into the component, then I will do that.

File: src/applications/gi/utils/render.jsx (hunk 2 above)
Comment (hughes-dustin) [reply src/applications/gi/utils/render.jsx]: Added in unit tests. I'll make adjustments to them once we hash how we want `renderLabel` to work.

**Review by cvalarida — APPROVED** [review]

**Pull request closed without merging**